packages/teams-js/test/public/mail.spec.ts

```diff
@@ -0,0 +1,325 @@
+          ),
+        );
+    });
+
+    it('should not allow calls from sidePanel context', async () => {
+      await utils.initializeWithContext('sidePanel');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "sidePanel".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from stage context', async () => {
+      await utils.initializeWithContext('stage');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "stage".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from meetingStage context', async () => {
+      await utils.initializeWithContext('meetingStage');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "meetingStage".'),
+          ),
+        );
+    });
+
+    it('should not allow calls if runtime does not support mail', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: {} });
+
+      await mail.openMailItem(openMailItemParams).catch(e => expect(e).toBe('Not Supported'));
+    });
+
+    it('should successfully throw if the openMailItem message sends and fails', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      const openMailItemPromise = mail.openMailItem(openMailItemParams);
+
+      const openMailItemMessage = utils.findMessageByFunc('mail.openMailItem');
+
+      const data = {
+        success: false,
+        error: 'Something went wrong...',
+      };
+
+      utils.respondToMessage(openMailItemMessage, data);
+      await openMailItemPromise.catch(e => expect(e).toMatchObject(new Error('Something went wrong...')));
+    });
+
+    it('should successfully send the openMailItem message', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      const promise = mail.openMailItem(openMailItemParams);
+
+      const openMailItemMessage = utils.findMessageByFunc('mail.openMailItem');
+
+      const data = {
+        success: true,
+        error: 'Something went wrong...',
+      };
+
+      utils.respondToMessage(openMailItemMessage, data);
+      await promise;
+
+      expect(openMailItemMessage).not.toBeNull();
+      expect(openMailItemMessage.args.length).toEqual(1);
+      expect(openMailItemMessage.args[0]).toStrictEqual(openMailItemParams);
```

Comment:
Since the only assertions here are on the message object received by the infrastructure, do you need to create a data object, respond to the message, and await the result? It seems like the thing you are testing don't require those actions to happen.

Reply:
This is a good question. I think the answer is that without responding to the message the promise from line 153 never finishes and the test times out.